File: 算法基础课(修订版)/笔试拿分技巧/P14161/user_acm.cc
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    long long solve(vector<int>& times, vector<pair<int, int>>& mutexPairs) {
        // 请在这里实现
        return 0;
    }
};

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int jobNum;
    cin >> jobNum;
    vector<int> times(jobNum);
    for (int i = 0; i < jobNum; ++i) {
        cin >> times[i];
    }

    int mutexNum;
    cin >> mutexNum;
    vector<pair<int, int>> mutexPairs(mutexNum);
    for (int i = 0; i < mutexNum; ++i) {
        cin >> mutexPairs[i].first >> mutexPairs[i].second;
    }

    Solution solution;
    cout << solution.solve(times, mutexPairs);
    return 0;
}

File: 算法基础课(修订版)/笔试拿分技巧/P14161/user_acm.java
import java.util.ArrayList;
import java.util.Scanner;

class Solution {
    public int solve(ArrayList<Integer> times, ArrayList<int[]> mutexPairs) {
        // 请在这里实现
        return 0;
    }
}

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int jobNum = sc.nextInt();
        ArrayList<Integer> times = new ArrayList<>();
        for (int i = 0; i < jobNum; i++) {
            times.add(sc.nextInt());
        }

        int mutexNum = sc.nextInt();
        ArrayList<int[]> mutexPairs = new ArrayList<>();
        for (int i = 0; i < mutexNum; i++) {
            mutexPairs.add(new int[] {sc.nextInt(), sc.nextInt()});
        }

        Solution solution = new Solution();
        System.out.println(solution.solve(times, mutexPairs));
        sc.close();
    }
}
